app/controllers/grant_submissions/submissions_controller.rb

```diff
@@ -24,46 +24,51 @@ def new
                               )
                     .find(params[:grant_id])
       @grant = GrantDecorator.new(@grant)
-      submission
+      set_submission
       authorize @submission
       render 'new'
     end
 
     def edit
       @grant = GrantDecorator.new(@grant)
-      submission
+      set_submission
       authorize @submission
       render 'edit'
     end
 
     def create
-      submission
+      set_submission
       authorize @submission
-      if @submission.save
+      set_state(@submission)
+      if @submission.save(context: @submission.state.to_sym)
         flash[:notice] = 'You successfully applied'
-        redirect_to grant_path(@grant)
+        submission_redirect(@grant, @submission)
       else
+        @submission.state = 'draft'
         @grant = GrantDecorator.new(@grant)
         flash.now[:alert] = @submission.errors.to_a
         render 'new'
       end
     end
 
     def update
-      submission
+      set_submission
       authorize @submission
-      if @submission.update(submission_params)
+      set_state(@submission)
+      @submission.assign_attributes(submission_params)
+      if @submission.save(context: @submission.state.to_sym)
         flash[:notice] = 'Submission was successfully updated'
-        #TODO: redirect based on user permissions
-        redirect_to grant_submissions_path(@grant)
+        submission_redirect(@grant, @submission)
       else
+        @submission.state = 'draft'
```

Comment:
the change to use button name/value instead of commit is great.
but this line seems weird. If the submission is already in a submitted state, and user clicks Submit, it should remain in Submitted state even if it doesn't pass validation and user has to change something. Any reason this is added?